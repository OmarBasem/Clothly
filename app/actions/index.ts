import {fetchUserAction} from "./user";
import {fetchProductsAction} from "./products";

export {fetchUserAction, fetchProductsAction}