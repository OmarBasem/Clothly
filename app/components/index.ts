import Title from "./Title";
import Back from './Back';
import TopBar from "./TopBar";
import SearchBar from "./SearchBar";
import Product from "./Product";
import ProductsRow from "./ProductsRow";


export {
    Title,
    Back,
    SearchBar,
    TopBar,
    Product,
    ProductsRow
}