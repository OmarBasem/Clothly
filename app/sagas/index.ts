import {all} from 'redux-saga/effects';

import user from './user'
import products from "./products";

const root = function* root() {
    yield all([
        user(),
        products()
    ]);
};

export default root;