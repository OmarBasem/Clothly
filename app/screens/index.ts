import HomeScreen from "./Home/HomeScreen";
import ProductScreen from "./Home/ProductScreen";
import CartScreen from "./Cart/CartScreen";
import ProfileScreen from "./Profile/ProfileScreen";
import WishlistScreen from "./Wishlist/WishlistScreen";

export {
    HomeScreen,
    ProductScreen,
    CartScreen,
    ProfileScreen,
    WishlistScreen
}