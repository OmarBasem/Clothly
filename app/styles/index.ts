import {DefaultTheme} from "@react-navigation/native";

export const theme = {
    mainColor: '#1C284F'
}


export const navTheme = {
    ...DefaultTheme,
    colors: {
        ...DefaultTheme.colors,
        background: '#fff'
    },
};
