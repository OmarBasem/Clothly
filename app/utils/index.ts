import {Dimensions} from "react-native";

export const sliceArrIntoRows = (data: Array<any>, n: number) => {
    const result = [];
    let temp = [];
    for (let i = 0; i < data.length; ++i) {
        if (i > 0 && i % n === 0) {
            result.push(temp);
            temp = [];
        }
        temp.push(data[i]);
    }
    if (temp.length > 0) {
        while (temp.length !== n) {
            temp.push(null);
        }
        result.push(temp);
    }
    return result;
};

export const W = (percentage: number) => Dimensions.get('window').width / 100 * percentage;
export const H = (percentage: number) => Dimensions.get('window').height / 100 * percentage;