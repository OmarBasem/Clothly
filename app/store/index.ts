import {configureStore} from '@reduxjs/toolkit'
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist'
import rootReducer from '../reducers'
import AsyncStorage from '@react-native-async-storage/async-storage';
import createSagaMiddleware from 'redux-saga'
import sagas from "../sagas";

const persistConfig = {
    key: 'root',
    storage: AsyncStorage,
    whitelist: [] // persist the user data in the redux store
}

const persistedReducer = persistReducer(persistConfig, rootReducer)
const sagaMiddleware = createSagaMiddleware()

export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
            },
        }).concat(sagaMiddleware),
})
sagaMiddleware.run(sagas)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export const persistor = persistStore(store)


