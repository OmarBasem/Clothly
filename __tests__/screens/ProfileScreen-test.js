import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store'
import renderer from "react-test-renderer";
import {ProfileScreen} from '../../app/screens'

const testUser = require('../data/testUser.json')


describe('Testing Screens rendering', () => {

    test('ProfileScreen renders correctly', () => {
        const mockStore = configureStore()
        const store = mockStore({user: testUser})
        const tree = renderer.create(<Provider store={store}><ProfileScreen /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})