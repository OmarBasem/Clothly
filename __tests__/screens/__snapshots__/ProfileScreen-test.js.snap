// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Screens rendering ProfileScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
  testID="ProfileScreen"
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "marginTop": 16,
          "padding": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "alignSelf": "center",
              "borderRadius": 150,
              "height": 300,
              "marginBottom": 8,
              "width": 300,
            },
          ]
        }
      >
        <FastImageView
          resizeMode="cover"
          source={
            Object {
              "uri": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Account Info
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderColor": "lightgrey",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginTop": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          First Name: 
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Morty
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          Last Name: 
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Smith
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          Gender: 
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Male
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          Location: 
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Citadel of Ricks
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          Status: 
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Alive
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          Created: 
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            11
             / 
            2017
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "marginTop": 16,
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Your Orders
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderColor": "lightgrey",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginTop": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          You have no recent orders.
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginTop": 28,
          "opacity": 1,
          "padding": 8,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Logout
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
